### PR TITLE
pvCopy: implement plugin filters

### PR DESCRIPTION
**Pull request by mrkraimer**
mplement plugin filters for copy facility.
This is modeled after the plugin filters that were introduced in epics base 3.15.

### CODE REVIEW

File: src/org/epics/pvdata/copy/arrayPlugin/ArrayFilter.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            return null;
+        }
+        int start =0;
+        int increment =1;
+        int end = -1;
+        String[] values = requestValue.split(":");
+        int num = values.length;
+        if(num==1) {
+            String value = values[0];
+            try{
```
Comment (duchuyctlk) [inline src/org/epics/pvdata/copy/arrayPlugin/ArrayFilter.java]: This `try-catch` block is used multiple times. Please consider extract it into a method.

**Comment by kasemir** [comment]
Isn't the alarm message a String, so this
```
if(current.getMessage()!=alarm.getMessage()) 
```
should be
```
if (!current.getMessage().equals(alarm.getMessage())) 
```
or in case there could be nulls,
```
if (! Objects.equals(current.getMessage(), alarm.getMessage())) 
```
?

**Comment by mrkraimer** [comment]
Kay is correct.
It also is a problem with the changes I just made to PVAlarm, and PVDisplay.
We should discuss at todays hangout.
Kay please join the hangout.

**Comment by ralphlange** [comment]
If I remember correctly, earlier this year we agreed to not further develop this idea until there is a user request for it.

**Comment by mrkraimer** [comment]
I am closing this pull request and will issue a new one based on decisions made at  Oct 17, 2017 hangout

**Pull request closed without merging**